app/scripts/components/data-catalog/index.tsx

```diff
@@ -189,6 +190,8 @@ function DataCatalog({ datasets }: DataCatalogProps) {
   const search = controlVars.search ?? '';
   let urlTaxonomyItems: OptionItem[] = [];
 
+  const datasetTaxonomies = generateTaxonomies(allDatasets);
```

Comment:
🔧 : The work surrounding creating the data layer abstraction is around making sure that the data layer remains our single source of truth. `allDatasets` is already doing that because its pulling `veda_datasets` from the data-layer which I had moved in the scaffolding PR. But another part that is important is that in the `data-utils` file we should have these utility methods take in the datasets as a param instead of using datasets directly (made a note of this in code). 

This is because it makes a bit more sense in my opinion to type these methods to expect a certain data structure rather than using the data directly, this is so we can create some abstraction to use these methods more flexibly when we get to pass in the data ourselves. Also with the work around creating the data-layer abstraction, we want the datasets to be provided to the core feature and the core feature should work with the dataset it was explicitly given (hence the data should be provided to these methods directly). We will have to update these methods, I made a note of it but didn't get the chance to do it myself with the scaffolding pr because it was a little out of scope.

Reply:
Update made. Updated `data-catalog/container` to instead provide `allDatasets` from `exploration/data-utils` (prev it provided `allDatasetsWithEnhancedLayers` which used `allDatasets` directly). Then updated `allDatasetsWithEnhancedLayers` to be a method instead that takes in the dataset as a param and updated data-catalog to support these changes. 

This makes sure we provide the expected dataset to data-catalog and data-catalog will use the utils to manipulate the dataset explicitly given to it.